src/averageAges.js

```diff
@@ -15,13 +15,22 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+  const result = people.filter(person => !century
+    ? person.sex === 'm'
+    : Math.ceil(person.died / 100) === century && person.sex === 'm'
+  );
+
+  const total = result.reduce((sum, person) =>
+    sum + (person.died - person.born),
+  0);
 
+  return total / result.length;
```

Comment:
move it to the separate function, and use it here and in the next function

Reply:
you are doing almost the same three times (but in the third function you should prepare data before reducing) - so better to move it to a separate function as suggested

I'll approve, but please, in the future take into account all comments (if something is unclear, etc - feel free to ask in chat)